curriculum/challenges/english/25-front-end-development/lecture-working-with-css-flexbox/672bd658c0e190f674a5e057.md

```diff
@@ -8,7 +8,124 @@ dashedName: what-are-some-common-flex-properties
+
+`flex` properties are properties that you can apply to `flex` containers to determine the distribution of child elements. We’ll cover some of the most commonly used ones: `flex-wrap`, `justify-content`, and `align-items`.
+
+Let’s start with `flex-wrap`. This property determines how `flex` items are wrapped within a `flex` container to fit the available space. `flex-wrap` can take three possible values: `nowrap`, `wrap`, and `wrap-reverse`. `nowrap` is the default value—`flex` items won’t be wrapped onto a new line, even if their `width` exceeds the container's `width`.
+
+In the code below, we have three `div` elements. Let’s focus on the `width`. The `main` container with the `red` `border` has a `width` of `200px`, while its three child `div` elements combined have a `width` of `240px` (`80px` each):
+
+```css
+main {
+  width: 200px;
+  display: flex;
+  border: 2px solid red;
+}
+
+div {
+  width: 80px;
+  height: 50px;
+}
+
+#first-div {
+  background-color: #4d70b2;
+}
+
+#second-div {
+  background-color: #5c4db2;
+}
+
+#third-div {
+  background-color: #4da3b2;
+}
+```
+
+The `width` of the `div` elements exceeds the `width` of their container, but by default they will be shrunk to fit the available space. If you do want to wrap them when they exceed the `width` of their container, you can set `flex-wrap: wrap` on the `flex` container:
+
+```css
+flex-wrap: wrap;
+```
+
+The `div` elements will be rearranged in rows when they exceed the `width` of their container. You can wrap `flex` items in reverse order with `flex-wrap: wrap-reverse`:
+
+```css
+flex-wrap: wrap-reverse;
+```
+
+The `flex-flow` property is a shorthand property for `flex-direction` and `flex-wrap`. In this example, we set `flex-direction` to `column` and `flex-wrap` to `wrap-reverse`:
+
+```css
+flex-flow: column wrap-reverse;
+```
+
+Great. Now let’s talk about `justify-content`. `justify-content` aligns the child elements along the main axis of the `flex` container. If you assign the value `flex-start` to `justify-content`, the `flex` items will be aligned to the start of the main axis. This could be horizontal or vertical:
+
+```css
+justify-content: flex-start;
+```
+
+With `justify-content: flex-end`, `flex` items are aligned to the end of the main axis, horizontally or vertically:
+
+```css
+justify-content: flex-end;
+```
+
+To center the `flex` items along the main axis, you can use `justify-content: center`:
+
+```css
+justify-content: center;
+```
+
+To distribute the elements evenly along the main axis, you can use `justify-content: space-between`. This will add some space between the `flex` items if needed:
+
+```css
+justify-content: space-between;
+```
+
+`justify-content: space-around` distributes `flex` items evenly within the main axis, adding a space before the first item and after the last item. This additional space is half of the space between each pair of adjacent items. If there’s only one item to distribute, it will be centered:
```

Comment:
```suggestion
`justify-content: space-around` distributes flex items evenly within the main axis, adding a space before the first item and after the last item. This additional space is half of the space between each pair of adjacent items. If there's only one item to distribute, it will be centered:
```